dotcom-rendering/src/web/components/EnhancePinnedPost.importable.tsx

```diff
@@ -42,40 +43,48 @@ function scrollOnCollapse() {
 	}
 }
 
+const handleClickTracking = () => {
+	if (pinnedPostCheckBox instanceof HTMLInputElement) {
+		if (pinnedPostCheckBox.checked) {
+			submitComponentEvent({
+				component: {
+					componentType: 'LIVE_BLOG_PINNED_POST',
+					id: pinnedPost?.id,
+				},
+				action: 'CLICK',
+				value: 'show-more',
+			});
+		} else {
+			submitComponentEvent({
+				component: {
+					componentType: 'LIVE_BLOG_PINNED_POST',
+					id: pinnedPost?.id,
+				},
+				action: 'CLICK',
+				value: 'show-less',
+			});
+			scrollOnCollapse();
+		}
+	}
+};
+
 export const EnhancePinnedPost = () => {
+	const [hasBeenSeen, setHasBeenSeen] = useState(false);
+	const [isInView] = useIsInView({
+		threshold: 0.1,
+		repeat: true,
+		node: pinnedPost ?? undefined,
+	});
+
+	const pinnedPostTiming =
+		useRef<{ start: () => void; end: () => number; clear: () => void }>();
```

Comment:
```suggestion
	const pinnedPostTiming =
		useRef<typeof initPerf>();
```

Reply:
It didn't let me use the `useRef<typeof initPerf>();` I'm afraid :( . I think it's because the `initPerf` is a function and `typeof initPerf` is giving the type of the function rather than the return type of the function

Reply:
Sorry–It’s probably [`ReturnType<typeof initPerf>`](https://www.typescriptlang.org/docs/handbook/utility-types.html#returntypetype)

Reply:
Fantastic, it works now. I learnt so much stuff in this PR 🤩